### PR TITLE
Notifications: Fix Yes/No condition on non english languages

### PR DESCRIPTION
**Pull request by AdrienClairembault**
Internal ref: 19319.

Notifications allow boolean condition, e.g. `##IFfollowup.isprivate=No##`.
This conditions are only working for the English language because the data we are comparing to is already translated to the value that will be used in the notification.

For example, if the target user of the notification is French and the followup is not private then the value will be "Non" -> "Non" != "No" so the condition will fail.

These changes fixes this issue by adapting the value used in the condition if needed.

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

### CODE REVIEW

**Comment by AdrienClairembault** [comment]
Not sure why tests are failing only for 7.4 + mariadb ?

**Comment by cedric-anne** [comment]
> Not sure why tests are failing only for 7.4 + mariadb ?

Could you rebase and force-push to trigger a new build ?

File: inc/notificationtemplate.class.php [hunk 1]
```diff
@@ -430,20 +434,24 @@ static function processIf($string, $data) {
...
+                  // Data value: the value for the field in the database
+                  $data_value = Html::entity_decode_deep($data['##'.$if_field.'##']);
+
+                  // Condition value: the expected value needed to validate the condition
+                  $condition_value = Html::entity_decode_deep($out[2][$key]);
+
+                  // Special case for data returned by Dropdown::getYesNo, we
+                  // need to use the localized value in the comparison
+                  $to_translate = ["Yes", "No"];
+                  if (in_array($condition_value, $to_translate)) {
```
Comment (cedric-anne) [inline inc/notificationtemplate.class.php:449]: Problem may also occured on other fields (e.g. `$tmp['##task.status##']       = Planning::getState($task['state']);`).
So why not try to translate every value ?

By the way, I do not know if checking condition only on translated value is a good point. Maybe check should be done on both based and translated value.

Comment (AdrienClairembault) [reply inc/notificationtemplate.class.php:449]: > By the way, I do not know if checking condition only on translated value is a good point. Maybe check should be done on both based and translated value.

Correct me if I'm wrong but with the current notification model I'm not sure we have access to the raw value ?
Obviously the best way would be to compare directly on these raw values but without reworking our current model it does not seem possible.

Comment (cedric-anne) [reply inc/notificationtemplate.class.php:449]: I was talking about comparing value with the translated one and the not translated one, i.e. the key 'Yes' and the translated 'Oui' in french.

Comment (cedric-anne) [reply inc/notificationtemplate.class.php:449]: By the way, is this `Yes` value an untranslated value or a value translated in the default language defined in GLPI installation process ? If it comes from the default language, your fix may only work in specific installations.

Comment (AdrienClairembault) [reply inc/notificationtemplate.class.php:449]: The values are translated to the current language:

```php
$tmp['##followup.isprivate##']   = Dropdown::getYesNo($followup['is_private']);
```


```php
static function getYesNo($value) {
  if ($value) {
     return __('Yes');
   }
   return __('No');
}
```

**Comment by AdrienClairembault** [comment]
FYI, internal ref tested and validated the fix work as expected for them.

**Pull request merged**